.github/workflows/ci-aot.yml

```diff
@@ -0,0 +1,33 @@
+name: Publish AOTCompatibility testApp
+
+on:
+  push:
+    branches: [ 'main*' ]
+    paths-ignore:
+    - '**.md'
+  pull_request:
+    branches: [ 'main*' ]
+    paths-ignore:
+    - '**.md'
+
+jobs:
+  aot-test:
+    strategy:
+      fail-fast: false  # ensures the entire test matrix is run, even if one permutation fails
+      matrix:
+        os: [ windows-latest ]
+        version: [ net6.0, net7.0 ]
+
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0 # fetching all
+
+    # powershell does not support AOT compiler yet
+    - name: Publich AOT testApp
+      run: dotnet publish ./test/OpenTelemetry.AotCompatibility.TestApp/OpenTelemetry.AotCompatibility.TestApp.csproj --self-contained -nodeReuse:false /p:UseSharedCompilation=false > ./build/publishAotTestAppOutput.log
```

Comment:
We should be able to remove `--self-contained` in this command. And instead make the .csproj so you can just `dotnet publish` it. For .NET 7, you need to put:

```xml
<SelfContained>true</SelfContained>
```

into the .csproj.

When we move to the .NET 8 SDK, we can remove that line because `<PublishAot>` implies `<SelfContained>` in the new version.